### PR TITLE
feat(ci): promote images from staging when deploying to prod

### PR DESCRIPTION
**Pull request by jfoo1984**
# Pull Request
[CZID-8804]

## Description
Implement prod deployment by promoting images from staging to prod.  Locks our deploy action to use the version containing chanzuckerberg/github-actions/pull/228, which enables a new argument needed for this feature.

[CZID-8804]: https://czi-tech.atlassian.net/browse/CZID-8804?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

## Notes

* Set `IMAGE_SOURCE_ROLE_ARN` as Github secret in production environment
* `IMAGE_SOURCE_ENV` and `IMAGE_SOURCE_STACK` were previously set.

### CODE REVIEW

File: .github/workflows/deploy-prod.yml [hunk 1]
```diff
@@ -38,10 +38,12 @@ jobs:
         run: |
           HAPPY_OIDC_ID_TOKEN=${{env.id_token}} happy config set API_URL ${{ vars.API_URL }} --env prod --aws-profile ""
       - name: Deploy to production env
-        uses: chanzuckerberg/github-actions/.github/actions/deploy-happy-stack@main
+        uses: chanzuckerberg/github-actions/.github/actions/deploy-happy-stack@v1.26.0
         with:
           tfe-token: ${{ secrets.TFE_TOKEN }}
           env: ${{ vars.HAPPY_ENV }}
-          create-tag: true
```
Comment (vincent-czi) [inline .github/workflows/deploy-prod.yml:45]: I see we're dropping `create-tag: true` entirely with this PR. I don't really know the specifics of `deploy-happy-stack`, but is this something we'd want to keep with what this PR is adding? It feels like if we wanted a tag created before, we would still want one now, but the new args don't seem to have anything to do with tagging one way or the other. Just wanted to make sure this was purposeful and not an accidentally cut out line!

**Review by vincent-czi — APPROVED** [review]
Seems reasonable! Long term, it would be nice to eventually unpin and get back to being on `@main` rather than staying pinned to a specific version of the deploy-happy-stack action, just in case there's eventually a security patch or something we might forget about.

**Comment by jfoo1984** [comment]
> Long term, it would be nice to eventually unpin and get back to being on @main rather than staying pinned to a specific version of the deploy-happy-stack action, just in case there's eventually a security patch or something we might forget about.

I believe the happy team actually recommended that I pin to a version, because they make make breaking changes, which would break our deploy pipelines if we automatically use the latest version

File: .github/workflows/deploy-prod.yml (hunk 1 above)
Comment (jfoo1984) [reply .github/workflows/deploy-prod.yml:45]: `create-tag` defaults to false

https://github.com/chanzuckerberg/github-actions/blob/0b740d81436af68cb9af1011cab89199823c0161/.github/actions/deploy-happy-stack/action.yml#L11-L14

To use the image promotion, `create-tag` does need to be set to false, so I've just dropped it here and relied on the default.

**Pull request merged**